src/util/markdown.js

```diff
@@ -0,0 +1,37 @@
+import memoize from 'lodash/memoize';
+import remark from 'remark';
+import remarkReact from 'remark-react';
+import remarkLowlight from 'remark-react-lowlight';
+import css from 'highlight.js/lib/languages/css';
+import xml from 'highlight.js/lib/languages/xml';
+import js from 'highlight.js/lib/languages/javascript';
+import githubSchema from 'hast-util-sanitize/lib/github.json';
+
+const remarkWithHighlighting = memoize(() => {
+  const schema = Object.assign({}, githubSchema, {
+    attributes: Object.assign({}, githubSchema.attributes, {
+      code: [
+        ...(githubSchema.attributes.code || []),
+        'className',
+      ],
+    }),
+  });
+
+  return remark().use(remarkReact, {
+    sanitize: schema,
+    remarkReactComponents: {
+      code: remarkLowlight({css, js, xml}),
+    },
+  });
+});
+
+const remarkWithoutHighlighting = memoize(() =>
+  remark().use(remarkReact),
+);
+
+export default {
```

Comment:
I’m generally not a fan of `export default`ing an object of functions—seems like named exports are a more idiomatic way to accomplish the same thing. How do you think about it?

Reply:
This is probably a holdover from other languages, but I like being able to use the module name to qualify a function... if I want to be able to write `markdown.toReact` I can use `import * as markdown from ...` (doesn't need to be a `default`) or `import markdown from ...` (needs to be a `default`).

I'd be fine switching to the `import * as markdown from ...` version, or abandoning the module qualification thing and changing the declaration to `export function markdownToReact` but those seem a little less natural. Up to you.

Reply:
mm, I definitely feel you and I like having the qualifier. Ultimately I don’t have a specific preference to stake out but a couple of observations:

* It seems like `import * as markdown` is a strict improvement, since you’re still getting the same interface in the importing module, and the exporting module structure is more idiomatic
* It doesn’t seem very common to do e.g. `import {toReact as markdownToReact}` but maybe that is the actual right answer? `markdownToReact` is just `markdown.toReact` without a dot, right?